neo/Prompt/Commands/Tokens.py

```diff
@@ -131,32 +138,114 @@ def command_desc(self):
         return CommandDesc('send', 'send a token from the wallet', [p1, p2, p3, p4, p5])
 
 
-def token_send(wallet, token_str, send_from, send_to, amount, prompt_passwd=True, user_tx_attributes=None):
+class CommandTokenSendFrom(CommandBase):
+    """
+    This command is for old style NEP-5 tokens before the proposal got amended to remove this optional command.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        wallet = PromptData.Wallet
+
+        if len(arguments) < 4:
+            print("Please specify the required parameters")
+            return False
+
+        if len(arguments) > 5:
+            # the 5th argument is prompting for a password before sending,
+            print("Too many parameters supplied. Please check your command")
+            return False
+
+        token_str = arguments[0]
+        from_addr = arguments[1]
+        to_addr = arguments[2]
+
+        try:
+            amount = float(arguments[3])
+        except ValueError:
+            print(f"{arguments[3]} is not a valid amount")
+            return False
+
+        prompt_passwd = True
```

Comment:
When we rebuild `Send.py`, we eliminated this argument.
https://github.com/CityOfZion/neo-python/blob/e3780126f9cf8175f9f3c7ad27f36f2baae20508/neo/Prompt/Commands/Send.py#L165-L169
Is there ever a case where the user should not be prompted? Also, in the case where other programs are using this function, they could use `mock` like I have implemented here: https://github.com/CityOfZion/neo-privatenet-docker/pull/63/files#diff-fb19bbb688a846c3107e299fceb63240R138

I recommend simplifying the code and removing `prompt_passwd` and allow dependent descriptions, etc.

Reply:
> Is there ever a case where the user should not be prompted?

Well, if the user is very confident in his command typing he could choose to run something potentially dangerous without confirmation. 

> Also, in the case where other programs are using this function, they could use mock like I have implemented here: https://github.com/CityOfZion/neo-privatenet-docker/pull/63/files#diff-fb19bbb688a846c3107e299fceb63240R138

`mock` is a unit testing library, using it for other purposes is mostly a hack to work around inflexible code. What you've linked kind of proves that the  `process_transaction()` function is inflexible. That function (like many I've recently encountered) mixes business logic with presentation logic (`prompt` actions). The easiest fix for `process_transaction()` would be to add an optional parameter whether to prompt for a password. Then your mocking isn't needed anymore.

Reply:
Okay, if that is preferred then I will add `prompt_passwd` back into `Send.py`.